engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java

```diff
@@ -259,9 +283,18 @@ private T loadCacheFrom(TruffleFile cacheRoot, EnsoContext context, TruffleLogge
       ByteBuffer blobBytes;
       var threeMbs = 3 * 1024 * 1024;
       if (file.exists() && file.length() > threeMbs) {
-        logger.log(Level.FINEST, "Cache file " + file + " mmapped with " + file.length() + " size");
-        var raf = new RandomAccessFile(file, "r");
-        blobBytes = raf.getChannel().map(FileChannel.MapMode.READ_ONLY, 0, file.length());
+        logger.log(
+            Level.FINEST,
+            "Cache file {0} mmapped with {1} size",
+            new Object[] {file, file.length()});
+        try (var chan = FileChannel.open(file.toPath())) {
+          assert memoryArena.scope().isAlive();
```

Comment:
- the previous `Arena` (if any) should be closed
- new `Arena` allocated and used for _mmapping_ the file

Reply:
In 72478c03585ecfc79a41d8c27df902741a18cf42 : 
- Previous arena is closed in `load` and `save`
- Before mmap, a new arena is created